crates/oxc_parser/src/js/list.rs

```diff
@@ -255,7 +255,8 @@ impl<'a> SeparatedList<'a> for FormalParameterList<'a> {
             _ => p.parse_binding_element()?,
         };
 
-        let decorators = p.state.consume_decorators();
+        let decorators =
+            p.state.consume_decorators().unwrap_or_else(|| Vec::new_in(p.ast.allocator));
```

Comment:
` consume_decorators` and `ParserState::decorators` should also be just a `Vec` at this point I think.

Reply:
In that case, should `consume_decorators` clear the array after taking the value?

Also, without the wrapping `Option` on `ParserState::decorators`, we can't automatically derive `Default`. Any suggestions there? 😄

Reply:
Managed to satisfy the compiler in this commit: https://github.com/Boshen/oxc/pull/84/commits/4ab68bdb4cc6f0ec00cec2a8b9c2501f770d14d4

However, still missing the implementation of `consume_decorators`, let me know what you're expecting there.

Reply:
Try `https://doc.rust-lang.org/std/mem/fn.take.html`, we want to take the value and replace it with an empty vec.

Reply:
Also try https://docs.rs/bumpalo/latest/bumpalo/collections/vec/struct.Vec.html#method.take or `drain` if it doesn't work because it's a bumpalo vector

Reply:
`take` only works on a slice and it's an experiment API so I don't think we can take advantage of it here. I also looked into `drain` but it returns `Drain<T>` and I can't figure out how to go from `Drain<T>` &rarr; `Vec<'a, Decorator<'a>>`. `.drain().collect()` seems to be missing implementation of `FromIterator<Decorator>` 🧐

Reply:
This should work

```rust
use std::collections::HashSet;

use oxc_allocator::{Allocator, Vec};
use oxc_ast::ast::Decorator;

pub struct ParserState<'a> {
    allocator: &'a Allocator,

    pub not_parenthesized_arrow: HashSet<u32>,

    pub decorators: Vec<'a, Decorator<'a>>,
}

impl<'a> ParserState<'a> {
    pub fn new(allocator: &'a Allocator) -> Self {
        Self {
            allocator,
            not_parenthesized_arrow: HashSet::new(),
            decorators: Vec::new_in(allocator),
        }
    }

    pub fn consume_decorators(&mut self) -> Vec<'a, Decorator<'a>> {
        std::mem::replace(&mut self.decorators, Vec::new_in(self.allocator))
    }
}
```